### ISSUE TITLE
Pokemon Main Series (XY, ORAS, SM, USUM) - Hangs When Soft Resetting

### ISSUE THREAD

**Issue by Leo626** [citra-emu/citra#3703]
Just want to add this here so that it won't be forgotten. The games hang whenever attempting to soft reset. The workaround to this is to exit the game and restart it. This can be done by Holding L+R+Start/Select, exiting the Mystery Gift menu, deleting save data (Up+X+B) or completing the game. The log shows this when it happens:

```
[ 11.317495] Kernel.SVC <Info> core/hle/kernel/svc.cpp:ExitProcess:147: Process 10 exiting
```

[Sun log](https://pastebin.com/a3EqFc8F)

Note: Pokemon X does not show this line but still hangs like the rest. Instead it shows this:

```
[ 22.290782] HW.Memory <Error> core/memory.cpp:GetPhysicalPointer:292: MMIO mappings are not supported yet. phys_addr=0x10B00000
[ 22.290788] HW.GPU <Critical> core/hw/gpu.cpp:MemoryFill:88: invalid end address 0X10B00000
[ 22.305937] Service <Error> core/hle/service/service.cpp:ReportUnimplementedFunction:163: unknown / unimplemented function 'ReplySleepQuery': port='APT:U' cmd_buf={[0]=0x3e0080, [1]=0x300, [2]=0x0}
```

[X log](https://pastebin.com/JAyLaJhW)

Tested with 9c65a45.

**Comment by CrisBRM** [comment]
Yup. Just completed pokemon US and this happened.

**Comment by aomaom880** [comment]
thx

**Comment by zhaowenlan1779** [comment]
My comment:
key point is this
```
[  11.315402] Service <Error> core/hle/service/service.cpp:ReportUnimplementedFunction:163: unknown / unimplemented function 'PrepareToDoApplicationJump': port='APT:A' cmd_buf={[0]=0x310100, [1]=0x2, [2]=0xffffffff, [3]=0xffffffff, [4]=0x0}
[  11.316197] Service.GSP <Warning> core/hle/service/gsp/gsp_gpu.cpp:ReleaseRight:698: called
[  11.316915] Service <Error> core/hle/service/service.cpp:ReportUnimplementedFunction:163: unknown / unimplemented function 'DoApplicationJump': port='APT:A' cmd_buf={[0]=0x320084, [1]=0x0, [2]=0x0, [3]=0x2, [4]=0x0, [5]=0x802, [6]=0x0}
```

**Comment by Dragios** [comment]
I think Pokémon X/Y soft reset is also broken due to audio. With DSP LLE I don not encounter any of this issue anymore. This issue can be close.

**Comment by zhaowenlan1779** [comment]
This should close after #4490 is merged.

**Comment by zhaowenlan1779** [comment]
Hmm, DSP LLE isn't merged yet, but it's kinda clear that Pokémon X/Y can do soft reset, just that it softlocked before resetting, and I think that's out of scope for this issue.

I'll close this for now. If anyone has a different idea, feel free to reopen this. Thanks!
